test/extended/util/client.go

```diff
@@ -748,14 +725,21 @@ func (c *CLI) Run(command string, flags ...string) *CLI {
 	if c.asPClusterKubeconf && !c.withoutNamespace {
 		FatalErr("you are doing something in ns of pcluster, please use WithoutNamespace and set ns in Args, for example, oc.AsPClusterKubeconf().WithoutNamespace().Run(\"get\").Args(\"pods\", \"-n\", \"pclusterNamespace\").Output()")
 	}
-	if !c.withoutNamespace && c.Namespace() != "" {
-		nc.flags = append(nc.flags, fmt.Sprintf("--namespace=%s", c.Namespace()))
+
+	if !c.withoutNamespace {
+		if !c.asPClusterKubeconf && c.isKCPCommand() && c.currentWorkSpace.CurrentNameSpace != "" {
```

Comment:
The KCP commands kubectl ws, kubectl kcp, kubectl workspaces seem to not have the flag --namespace. So line 730 should use `!c.isKCPCommand()` instead of `c.isKCPCommand()`?

Reply:
```
        if !c.withoutNamespace {
                if !c.isKCPCommand() && c.asPClusterKubeconf && c.Namespace() != "" {
                        nc.flags = append(nc.flags, fmt.Sprintf("--namespace=%s", c.Namespace()))
                }
                if !c.isKCPCommand() && !c.asPClusterKubeconf && c.currentWorkSpace.CurrentNameSpace != "" {
                        nc.flags = append(nc.flags, fmt.Sprintf("--namespace=%s", c.currentWorkSpace.CurrentNameSpace))
                }
        }
```